xarray/coding/times.py

```diff
@@ -363,24 +437,36 @@ def decode_cf_datetime(
 
 
 def to_timedelta_unboxed(value, **kwargs):
+    # todo: check, if the procedure here is correct
     result = pd.to_timedelta(value, **kwargs).to_numpy()
-    assert result.dtype == "timedelta64[ns]"
+    unique_timedeltas = np.unique(result[pd.notnull(result)])
+    unit = _netcdf_to_numpy_timeunit(_infer_time_units_from_diff(unique_timedeltas))
+    if unit not in ["s", "ms", "us", "ns"]:
+        unit = "s"
+    result = result.astype(f"timedelta64[{unit}]")
+    assert np.issubdtype(result.dtype, "timedelta64")
     return result
 
 
 def to_datetime_unboxed(value, **kwargs):
     result = pd.to_datetime(value, **kwargs).to_numpy()
-    assert result.dtype == "datetime64[ns]"
+    assert np.issubdtype(result.dtype, "datetime64")
     return result
 
 
 def decode_cf_timedelta(num_timedeltas, units: str) -> np.ndarray:
+    # todo: check, if this works as intended
     """Given an array of numeric timedeltas in netCDF format, convert it into a
-    numpy timedelta64[ns] array.
+    numpy timedelta64 ["s", "ms", "us", "ns"] array.
     """
     num_timedeltas = np.asarray(num_timedeltas)
-    units = _netcdf_to_numpy_timeunit(units)
-    result = to_timedelta_unboxed(ravel(num_timedeltas), unit=units)
+    unit = _netcdf_to_numpy_timeunit(units)
+    as_unit = unit
+    if unit not in ["s", "ms", "us", "ns"]:
+        as_unit = "s"
```

Comment:
same concern as above -- should this raise an error instead?

Reply:
Similar to above, but the other way round (decoding). `pd_to_timedelta` returns `ns` resolution. This makes sure the data is returned in the lowest possible resolution the intermediate `TimedeltaIndex` can handle. So for decoding "days" we finally get `s` resolution, which seems more appropriate than `ns`.